### PR TITLE
Add searching for instances

### PR DESCRIPTION
**Pull request by micahmo**
## Pull Request Description

This PR aims to make instance exploration easier, especially now that we have #1133, by adding them as a searchable type on the search page. To do this, I had to extend `SearchType` with `MetaSearchType`, which is a superset. When using this type, all other search filters are hidden. There is also the option to view all instances without searching. This "view all" technique could be expanded to other search types to satisfy [this comment](https://github.com/thunder-app/thunder/pull/1133#issuecomment-1967305567).

Note that I am filtering by instances running Lemmy software that have successfully federated in the past 7 days. When I didn't do this, there were so many dead instances that it was almost useless. Even with this, there are some instances that seem unreachable (and are designated as such), so we might have to tweak this.

~~**Note**: I may try to improve the jarring change when the metadata fills in for the instances. Using `AnimatedContainer`, at least the icon fades, but the text doesn't. Anyway, it's just a cosmetic thing for now.~~

> Review without whitespace.

## Issue Being Fixed

Issue Number: N/A

### Search for a particular instance

https://github.com/thunder-app/thunder/assets/7417301/5cfda56b-3c07-43b0-962d-e79ce0c10ec0

### View all instances

https://github.com/thunder-app/thunder/assets/7417301/5be03e18-a1d6-4f7e-a2d3-e4a577f8db44

## Checklist

- [ ] Did you update CHANGELOG.md?
- [ ] Did you use localized strings where applicable?
- [ ] Did you add `semanticLabel`s where applicable for accessibility?

### CODE REVIEW

**Comment by micahmo** [comment]
I've pushed a change which allows the metadata to populate in more gracefully.

https://github.com/thunder-app/thunder/assets/7417301/331935ad-d27f-439c-9534-b444dd7967e9

File: lib/search/pages/search_page.dart [hunk 1]
```diff
@@ -477,17 +482,26 @@ class _SearchPageState extends State<SearchPage> with AutomaticKeepAliveClientMi
...
                     },
                     textAlign: TextAlign.center,
                     style: theme.textTheme.titleMedium?.copyWith(color: theme.dividerColor),
                   ),
                 ),
               ],
+              if (_currentSearchType == MetaSearchType.instances && _controller.text.isEmpty) ...[
+                const SizedBox(height: 30),
+                SearchActionChip(
+                  // todo
```
Comment (hjiangsu) [inline lib/search/pages/search_page.dart:500]: What is the `todo` here?

**Review by hjiangsu — APPROVED** [review]
Looks fairly good to me! Only a couple of comments!

File: lib/l10n/app_en.arb [hunk 2]
```diff
@@ -615,6 +615,10 @@
   "@instance": {},
   "instanceHasAlreadyBenAdded": "{instance} has already been added.",
   "@instanceHasAlreadyBenAdded": {},
+  "instances": "Instances",
```
[the remark this answers is not in the document]
Comment (micahmo) [reply lib/l10n/app_en.arb]: Oh yes, good call! Done!

File: lib/search/pages/search_page.dart (hunk 1 above)
Comment (micahmo) [reply lib/search/pages/search_page.dart]: Oops, just a leftover. Removed!

**Comment by micahmo** [comment]
Should be all set now!

**Pull request merged**